File: src/MinCamlSharp.Compiler/CompilerOptions.cs
using System.IO;

namespace MinCamlSharp
{
	public class CompilerOptions
	{
		public string SourceFile { get; set; }
		public string OutputAssemblyName { get; set; }
		public string OutputFilePath { get; set; }

		public string OutputDirectory
		{
			get { return Path.GetDirectoryName(OutputFilePath); }
		}

		public string OutputFileName
		{
			get { return Path.GetFileName(OutputFilePath); }
		}
	}
}

File: src/MinCamlSharp.Compiler/CompilerOptions.fs
namespace MinCamlSharp.Compiler

open System.IO

type CompilerOptions(sourceFile : string, outputAssemblyName : string, outputFilePath : string) =
    let outputDirectory = Path.GetDirectoryName(outputFilePath)
    let outputFileName = Path.GetFileName(outputFilePath)

    member x.SourceFile = sourceFile
    member x.OutputAssemblyName = outputAssemblyName
    member x.OutputFilePath = outputFilePath
    member x.OutputDirectory = outputDirectory
    member x.OutputFileName = outputFileName